Stack/Opc.Ua.Core/Stack/Tcp/TcpListenerChannel.cs

```diff
@@ -158,6 +158,18 @@ public void Attach(uint channelId, Socket socket)
                 StartCleanupTimer(StatusCodes.BadTimeout);
             }
         }
+
+        /// <summary>
+        /// Set the channel Inactive and clean it up
+        /// </summary>
+        public void Cleanup()
+        {
+            lock (DataLock)
```

Comment:
there needs to be a way find out if a session is used by a session. If not, flag it inactive.

Reply:
you "mean if a channel is used by a session"?

Reply:
This lock here is unnecessary as the State is not a shared resource. It is just a status flag. - Probability of Inactive flag overridden is close to nil and even if it is overridden, it would not affect the end result. But keeping the lock would hamper performance.

Reply:
> This lock here is unnecessary as the State is not a shared resource. It is just a status flag. - Probability of Inactive flag overridden is close to nil and even if it is overridden, it would not affect the end result. But keeping the lock would hamper performance.

I think this lock is necessary because it protects a state transition. If there is an ongoing transition just setting the new state may cause chaos.